src/pages/resultsView/enrichments/AlterationEnrichmentsTable.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+import * as _ from "lodash";
+import LazyMobXTable, { Column } from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import { observer } from "mobx-react";
+import { observable } from "mobx";
+import { Badge, Checkbox } from 'react-bootstrap';
+import {
+    calculateAlterationTendency, formatLogOddsRatio, formatPercentage, formatValueWithStyle,
+    formatLogOddsRatioWithStyle
+} from "./EnrichmentsUtil";
+import { toConditionalPrecision } from 'shared/lib/NumberUtils';
+import styles from "./styles.module.scss";
+import { AlterationEnrichmentRow } from 'shared/model/AlterationEnrichmentRow';
+
+export interface IAlterationEnrichmentTableProps {
+    columns?: AlterationEnrichmentTableColumnType[];
+    data: AlterationEnrichmentRow[];
+    initialSortColumn?: string;
+    onCheckGene: (hugoGeneSymbol: string) => void;
+    onGeneNameClick: (hugoGeneSymbol: string) => void;
+}
+
+export enum AlterationEnrichmentTableColumnType {
+    GENE,
+    CYTOBAND,
+    PERCENTAGE_IN_ALTERED,
+    PERCENTAGE_IN_UNALTERED,
+    LOG_RATIO,
+    P_VALUE,
+    Q_VALUE,
+    TENDENCY
+}
+
+type AlterationEnrichmentTableColumn = Column<AlterationEnrichmentRow> & { order?: number, shouldExclude?: () => boolean };
+
+export class AlterationEnrichmentTableComponent extends LazyMobXTable<AlterationEnrichmentRow> {
+}
+
+@observer
+export default class AlterationEnrichmentTable extends React.Component<IAlterationEnrichmentTableProps, {}> {
+
+    @observable protected _columns: { [columnEnum: number]: AlterationEnrichmentTableColumn };
+
+    constructor(props: IAlterationEnrichmentTableProps) {
+        super(props);
+        this._columns = {};
+        this.generateColumns();
+    }
+
+    public static defaultProps = {
+        columns: [
+            AlterationEnrichmentTableColumnType.GENE,
+            AlterationEnrichmentTableColumnType.CYTOBAND,
+            AlterationEnrichmentTableColumnType.PERCENTAGE_IN_ALTERED,
+            AlterationEnrichmentTableColumnType.PERCENTAGE_IN_UNALTERED,
+            AlterationEnrichmentTableColumnType.LOG_RATIO,
+            AlterationEnrichmentTableColumnType.P_VALUE,
+            AlterationEnrichmentTableColumnType.Q_VALUE,
+            AlterationEnrichmentTableColumnType.TENDENCY
+        ],
+        initialSortColumn: "q-Value"
+    };
+
+    private checkboxChange(hugoGeneSymbol: string) {
+        const row: AlterationEnrichmentRow = _.find(this.props.data, {hugoGeneSymbol})!;
+        row.checked = !row.checked;
+        this.generateColumns();
+        this.props.onCheckGene(hugoGeneSymbol);
+    }
+
+    private geneNameClick(hugoGeneSymbol: string) {
+        this.props.onGeneNameClick(hugoGeneSymbol);
+    }
+
+    protected generateColumns() {
+        this._columns = {};
```

Comment:
make this a pure function.  it should return the columns object and calling code can assign it to the instance prop